geps/gep-1619.md

```diff
@@ -163,43 +163,258 @@ Java application servers such as Tomcat and Jetty, were the first to standardize
+application's architecture by the developers themselves. Consequently, application developers should be responsible for
+adjusting the configuration to ensure the functionality of their applications.
 
-A set-cookie request can be broken down into [3 main components](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie):
-- Cookie Name
-- Cookie Value
-- Cookie Attributes
+**What existing API object it is augmenting?**
 
-Only the Cookie Name and Cookie Value are required. All Cookie Attributes are optional. Attributes are simply key=value pairs with the value as optional for some attributes.
+Session persistence augments a Kubernetes service object after the routing to the service has taken place. In
+Kubernetes, a service is an abstract representation of a set of pods that work together to provide a specific
+functionality. Services perform load balancing, routing traffic from a service endpoint to a pod. Session persistence
+augments the load balancing functionality of the service by consistently directing traffic to the same pod.
 
-The set-cookies attributes defined by [rfc6265](https://www.rfc-editor.org/rfc/rfc6265#section-5.2) are:
+**What are the practical implications of attaching session persistence to an existing API object?**
+
+One significant practical implication of adding session persistence to services is its reusability across multiple
+routes. This reusability can be advantageous as it eliminates the need for repetitive session persistence configuration
+for each individual route. However, on the flip side, it also mandates the inclusion of session persistence
+configuration for every route that utilizes the service. If session persistence is a requirement of the application
+architecture, there is no need to configure it separately for each route.
+
+**How are existing implementation APIs designed?**
+
+Referring to our [Implementations](#Implementations) tables on session persistence, the majority of Gateway API
+implementations designed session persistence in their APIs to be attached to a service or backends. This should be
+considered cautiously, as making associations to Gateway API's notion of Gateway, Route, and Service to other
+implementation's objects is hard to directly translate. The idea of a route in Gateway API is often not the same as a
+route in any given implementation.
+
+Taking Istio as an example, it offers a clear attachment point of session persistence to a service through a [DestinationRule](https://istio.io/latest/docs/reference/config/networking/destination-rule/#DestinationRule).
+`DestinationRule` defines policies that apply to traffic intended for a service after routing has occurred. Within
+`DestinationRule`, Istio couples the selection of load balancing algorithms with session persistence via their [LoadBalancerSettings](https://istio.io/latest/docs/reference/config/networking/destination-rule/#LoadBalancerSettings).
```

Comment:
If you want to take into account traffic shifting/splitting (as mentioned in my previous comment), then the Istio object is [`VirtualService`](https://istio.io/latest/docs/reference/config/networking/virtual-service/) or `HttpRoute` in the new Gateway API

Reply:
Yes I agree, but here is talking quite literally. Istio's API attaches/augments a "service". Istio doesn't attach a destination rule configuration to a Virtual Service.

I think there is a difference here is in the literal vs. actual. Istio doesn't literally (i.e. API definitions) attach session persistence configuration to a Virtual Service, but I think your point is that, it actually (effectively, in practice) augments a Virtual Service in the use case of traffic splitting (note: i'm assuming that's what happens in practice, I've never tested this in Istio).

I think the argument for Gateway API design, is that we should _literally_ attach our Session Persistence API to the objects it will _actually_ augment. I.e. attach `SessionPersistencePolicy` to `HTTPRoute`.

This sounds better in my head and probably doesn't make much sense written out, but let me know if you can help me make sense of it.

Reply:
API semantics of attaching to route are just too complicated to understand and implement ( i.e. how different routes with or without persistence interact with each other ), and for a user who deploys a backend - persistence is a property of the implementation, not of how it is routed by different gateways ( a backend may be used in multiple gateways and in mesh ). 

Usability and semantics are simply not practical (beyond the trivial case where you have a single route). 

And I believe we have looked at all the existing systems - with only envoy doing things at route level - so for a common API I think we MUST support the higher granularity, with per-route optional for servers that can support it. I think that's consistent with the gateway extensibility model - many extensions can attach to different points.

Reply:
@costinm I take it that you are okay with attaching to xRoutes.

Reply:
I interpret @costinm comment to mean: attaching to Service (higher granularity) is at core support level, but attaching to a route is extended support level, but we should support both in the API. Let me know if I misunderstood.